dsl/camel-xml-jaxb-dsl-test/definition/src/test/java/org/apache/camel/dsl/xml/jaxb/definition/LoadRestFromXmlTest.java

```diff
@@ -54,10 +52,9 @@ public void testLoadRestsDefinitionFromXml() throws Exception {
         foo.assertIsSatisfied();
 
         // load rest from XML and add them to the existing camel context
-        InputStream is = getClass().getResourceAsStream("barRest.xml");
         ExtendedCamelContext ecc = context.adapt(ExtendedCamelContext.class);
-        RestsDefinition rests = (RestsDefinition) ecc.getXMLRoutesDefinitionLoader().loadRestsDefinition(ecc, is);
-        context.addRestDefinitions(rests.getRests(), true);
+        Resource resource = ecc.getResourceLoader().resolveResource("org/apache/camel/dsl/xml/jaxb/definition/barRest.xml");
+        ecc.getRoutesLoader().loadRoutes(resource);
```

Comment:
To avoid duplicating the same code, we could have a helper class that proposes a method allowing us to load the routes from a file of a given name located in a given package. This helper class could be added to the test classes of camel-core (the test classes of camel-core are packaged as a jar) to be reusable in different modules.

Reply:
I think this should be done in a separate commit if there's really a need to make a reusable method, as it would span a bunch of additional files.  
But if that's really something that should be reused, maybe a better place would be on the `ExtendedCamelContext` directly, as it would benefit end users too.